pom.xml

```diff
@@ -51,9 +51,23 @@ under the License.
         <developerConnection>scm:git:https://github.com/apache/flink-shaded.git</developerConnection>
     </scm>
 
+    <pluginRepositories>
+        <pluginRepository>
+            <!-- Allows exec-maven-plugin to resolve snapshot plugin dependencies -->
+            <id>apache.snapshots.https</id>
+            <name>${distMgmtSnapshotsName}</name>
+            <url>${distMgmtSnapshotsUrl}</url>
+        </pluginRepository>
+    </pluginRepositories>
+
     <properties>
         <shading.prefix>org.apache.flink.shaded</shading.prefix>
         <netty.version>4.1.70.Final</netty.version>
+        <!-- The license check requires the artifactId to match the directory that the module resides in.
+             This is not the case for several modules in flink-shaded for legacy reasons.
+             This property can be used by modules to add a suffix to the artifactId to match the directory name on CI
+             when the 'license-check' profile is active. -->
+        <flink.ci.license.suffix/>
```

Comment:
This thing is super jank but I don't have an idea on how to solve it in _general_ without potentially breaking _something_.